go/weed/weed_server/filer_server_handlers.go

```diff
@@ -134,14 +161,63 @@ func (fs *FilerServer) PostHandler(w http.ResponseWriter, r *http.Request) {
 	if collection == "" {
 		collection = fs.collection
 	}
-	assignResult, ae := operation.Assign(fs.master, 1, replication, collection, query.Get("ttl"))
-	if ae != nil {
-		glog.V(0).Infoln("failing to assign a file id", ae.Error())
-		writeJsonError(w, r, http.StatusInternalServerError, ae)
+
+	buf, _ := ioutil.ReadAll(r.Body)
+	r.Body = analogueReader{bytes.NewBuffer(buf)}
+	fileName, _, _, _, _, _, _, pe := storage.ParseUpload(r)
+	if pe != nil {
+		glog.V(0).Infoln("failing to parse post body", pe.Error())
+		writeJsonError(w, r, http.StatusInternalServerError, pe)
+		return
+	}
+	//reconstruct http request body for following new request to volume server
+	r.Body = analogueReader{bytes.NewBuffer(buf)}
```

Comment:
This would require the request be held in memory in filer during the process. Is it possible to avoid this or only do this if it is an update?